Classes/Service/DemandService.php

```diff
@@ -187,6 +231,91 @@ public function getInnerBoundsForProperty(string $propertyName): array
      */
     protected function getConfiguredDemandProviders(): array
     {
-        // TODO: Initialize DemandProvider objects configured in TypoScript and return in order of execution
+        $configurationManager = GeneralUtility::makeInstance(ConfigurationManager::class);
+        $config = $configurationManager->getConfiguration(ConfigurationManager::CONFIGURATION_TYPE_FULL_TYPOSCRIPT)['config.']['tx_demander.'];
+        $demandProviders = [];
+
+        foreach ($config['demandProviders.'] as $id => $demandProvider)
+        {
+            $demandProviders[$id] = GeneralUtility::makeInstance($demandProvider);
+        }
+        return $demandProviders;
+    }
+
+    /**
+     * Transform necessary values from string to integer.
+     *
+     * @param array $restrictionsArray
+     * @return array
+     */
+    protected function restrictionsToInt(array $restrictionsArray): array
+    {
+        $restrictions = [];
+
+        foreach ($restrictionsArray as $key => $restriction) {
+            if ($restriction['value']) {
+                $value = (is_numeric($restriction['value'])) ? (int)$restriction['value'] : $restriction['value'];
+                $restriction = array_replace($restriction, ['value' => $value]);
+                $restrictions[$key] = $restriction;
+            }else{
+                $restrictions[$key] = $this->restrictionsToInt($restriction);
+            }
+        }
+
+        return $restrictions;
+    }
+
+    /**
+     * Resolve field name from alias or from array if OR, AND provided.
+     *
+     * @param mixed $alias
+     * @param null|string $rootKey
+     * @return mixed
+     */
+    protected function getFieldPropsFromAlias($alias, $rootKey = null)
+    {
+        if ($rootKey !== null){
+            $rootKey = mb_substr($rootKey, 0, -1);
+            $fieldProps = [];
+
+            foreach ($alias as $key => $subAlias){
+                $fieldProps[$rootKey][$this->getFieldPropsFromAlias($key)] = array(
+                    $subAlias
+                );
+            }
+
+            return $fieldProps;
+        }
+
+        return mb_substr((explode('-',  $alias)[1]), 0, -1);
+    }
+
+    /**
+     * @param string $fieldname
+     * @param array $restrictions
+     * @param ExpressionBuilder $expressionBuilder
+     * @return string
+     */
+    protected function getExpressionFromRestrictions(string $fieldname, array $restrictions, ExpressionBuilder $expressionBuilder): string
```

Comment:
This might also be good to move into `DemandArrayUtility()`.

Also, this method isn't really a getter, but a converter, and it only handles a single restriction with operator and value(s). Consider something like this:

```suggestion
    protected function convertRestrictionToExpression(string $fieldname, array $restriction, ExpressionBuilder $expressionBuilder): string
```